### PR TITLE
Billing Manager and Some Module fixes

### LINKED ISSUE #7752: Module command line batch notifications broken.
### Describe the bug

Not loading rest dispatcher. I thought Robert had this working when he made some fixes a while back.
Anyway Broke, Broke and Broken!:)

I might run via repeater service as option in setup.

- Fix Billing Manager redirects

### Client configuration

- Browser:
- OpenEMR version: 702 all
- Operating system: all

### PR DESCRIPTION
**Pull request by sjpadgett**
fix billing managers toencounter and topatient.
check and set if passed in pid not current for encounter history

Minor stuff but thought provoking when looking into our tabs and patient view models!

Fixes #7752

### CODE REVIEW

File: interface/patient_file/history/encounters.php [hunk 1]
```diff
@@ -33,6 +33,9 @@
 
 $is_group = ($attendant_type == 'gid') ? true : false;
 
+if (isset($_GET['pid']) && $_GET['pid'] != $_SESSION['pid']) {
+    setpid($_GET['pid']);
+}
```
Comment (stephenwaite) [inline interface/patient_file/history/encounters.php:38]: almost there @sjpadgett!

getting an Unknown tab when switching between people when clicking on the Insurance button from the billing manager
Screenshot from 2024-10-01 16-18-31

Encounter button loads fine:
Screenshot from 2024-10-01 16-15-53

Insurance button issue:
Screenshot from 2024-10-01 16-19-30

Comment (sjpadgett) [reply interface/patient_file/history/encounters.php:38]: I switched database to one with modules and api enabled. I'm getting
```
[01-Oct-2024 16:29:23 America/New_York] PHP Fatal error:  Uncaught Error: Call to undefined function setpid() in C:\xampp\htdocs\openemr\interface\patient_file\history\encounters.php:37
Stack trace:
#0 {main}
  thrown in C:\xampp\htdocs\openemr\interface\patient_file\history\encounters.php on line 37
[01-Oct-2024 16:29:24 America/New_York] [2024-10-01T16:29:24.986858-04:00] OpenEMR.DEBUG: HttpRestRouteHandler::dispatch() start request {"resource":"","method":"GET","user":"","role":"","client":"","apiType":"oemr","route":"/api/patient/9353b8f5-0a87-4e2a-afd4-25341fdb0fbc/insurance","queryParams":[]} []
[01-Oct-2024 16:29:24 America/New_York] [2024-10-01T16:29:24.991857-04:00] OpenEMR.DEBUG: HttpRestParsedRoute->__construct() matched {"routePath":"GET /api/patient/:puuid/insurance","requestPath":"/api/patient/9353b8f5-0a87-4e2a-afd4-25341fdb0fbc/insurance","method":"GET","routeParams":["9353b8f5-0a87-4e2a-afd4-25341fdb0fbc"],"resource":"insurance","operation":""} []
[01-Oct-2024 16:29:24 America/New_York] [2024-10-01T16:29:24.993213-04:00] OpenEMR.DEBUG: HttpRestRouteHandler->dispatch() dispatching route {"route":"GET /api/patient/:puuid/insurance"} []
[01-Oct-2024 16:29:25 America/New_York] [2024-10-01T16:29:25.034661-04:00] OpenEMR.DEBUG: RestControllerHelper::handleFhirProcessingResult() Records found {"count":"3"} []

```

However, behavior is not consistent! Weird stuff. Man! Look what you did to me!:)
I'll look at a little later. I need to figure out why batch notifications aren't working from command line so I can sleep tonight.:)

Comment (sjpadgett) [reply interface/patient_file/history/encounters.php:39]: @stephenwaite This is working for me now. Let me know if you see any issues

**Comment by sjpadgett** [comment]
RuntimeException: "chromedriver" binary not found. Install it using the package manager of your operating system or by running "composer require --dev dbrekelmans/bdi && vendor/bin/bdi detect drivers".

File: interface/patient_file/history/encounters.php (hunk 1 above)
Comment (stephenwaite) [reply interface/patient_file/history/encounters.php:39]: ok, thanks, that fix makes sense, will check it out

**Comment by stephenwaite** [comment]
think Brady's working on that CI fail in https://github.com/openemr/openemr/pull/7739

**Review by stephenwaite — APPROVED** [review]
nice

**Comment by sjpadgett** [comment]
Do I need to merge master in PR for tests to update?

**Comment by sjpadgett** [comment]
I guess we'll see

**Comment by bradymiller** [comment]
yep, ignore that php 8.4 fail (hopefully get it fixed soon)

**Comment by adunsulag** [comment]
@sjpadgett does this need to go in rel-702?

**Comment by sjpadgett** [comment]
yep, any prs i've done since I last updated 702 from master will go back. I have one i'll do one tomorrow for appt comments and flow board then quit.

**Pull request merged**